docs/developing/deploy_facilities/interacting_with_spl_tokens.md

```diff
@@ -4,18 +4,90 @@ proofedDate: na
 ---
 
-The ERC-20 SPL wrapper contract provides access to native Solana tokens registered in the SPL token contract, through the ERC-20 interface.
+import Tabs from '@theme/Tabs';
+import TabItem from '@theme/TabItem';
 
-This allows Solana applications to interact with EVM (Solidity, Vyper, etc.) bytecode contracts. The ERC-20 SPL wrapper can also be used to transfer funds in Solana tokens using Ethereum wallets such as MetaMask.
+The ERC-20 SPL wrapper contract provides access to native Solana tokens, i.e. those registered in the SPL token contract, through the ERC-20 interface. 
 
-The contract is implemented in Rust as part of the Neon EVM program.
+This allows Solana liquidity to be available to EVM (Solidity, Vyper, etc.) bytecode contracts, i.e. this ERC-20 SPL wrapper allows Ethereum wallets such as MetaMask to transact with SPL tokens.
+
+:::info
+To be able to use an SPL token from a Solana account balance, it must be transferred to a NeonEVM account via the [NeonPass SDK](/docs/developing/integrate/neon_transfer_sdk).
+:::
+
+Two contracts are available on Neon EVM that enable SPL tokens to be transacted as ERC-20 compliant. It's vital that you understand which to use:
+
+## ERC-20-for-SPL
+The [ERC-20-for-SPL contract](https://github.com/neonlabsorg/neon-evm/blob/c33b34bb624234955d88bf98a4ad1c95ddd453dc/evm_loader/solidity/erc20_for_spl.sol#L12) works with a precompiled contract within Neon EVM which can call the SPL token program. This enables you to utilize existing SPL tokens e.g. SOL or NEON, as wSOL or wNEON, respectively, via the ERC-20 interface, i.e. this contract assigns the [ERC-20 standard](https://eips.ethereum.org/EIPS/eip-20) to the token.
+
+
+## ERC-20-for-SPL-Mintable
+The [ERC-20-for-SPL-Mintable](https://github.com/neonlabsorg/neon-evm/blob/4bcae0f476721e5396916c43396ec85e465f878f/evm_loader/solidity/erc20_for_spl.sol#LL203C1-L203C1) contract has two additional methods that enable you to use the Neon EVM to mint a new SPL token and wrap it as ERC-20-compatible. This contract creates a new SPL token using Solana's Token Program and provides mint and freeze authority to the Neon account specified in the constructor.
+
+| Contract            | Usage                                | Account requirements                               | Mint tx signed by                                                    |
+| ------------------- | ----------------------------------- | ------------------------------------------ | -------------------------------------------------------------------- |
+| ERC20ForSpl         | Wrap Solana-minted token | 1\. Neon Account<br></br>2\. Existing SPL token | Signed by the Solana account private key (via linked Phantom wallet) |
+| ERC20ForSplMintable | Mint a token on Solana   | 1\. Neon Account                           | Signed by the Neon account private key (via linked MetaMask wallet)  |
+
+
+
+<Tabs>
+ <TabItem value="Constructor non-mintable" label="ERC20ForSpl Constructor" default>
```

Comment:
Remove everything related to constructors.
Instead add link to documentation about erc20_for_spl_factory.

Reply:
Hmm -- this was requested by Semen due to the slight variation in the data requirements by the constructors

Let's add this to Huddle

Reply:
https://github.com/neonlabsorg/neon-evm/blob/4bcae0f476721e5396916c43396ec85e465f878f/evm_loader/solidity/erc20_for_spl_factory.sol Is the main contract that calls these sub contracts